packages/kit/src/runtime/server/page/csp.js

```diff
@@ -0,0 +1,197 @@
+import { escape_html_attr } from '../../../utils/escape.js';
+import { sha256, base64 } from './crypto.js';
+
+/** @type {Promise<void>} */
+export let csp_ready;
+
+/** @type {() => string} */
+let generate_nonce;
+
+/** @type {(input: string) => string} */
+let generate_hash;
+
+if (typeof crypto !== 'undefined') {
```

Comment:
Is it safe to assume existing crypto is web-crypto or should it do additional checks?

Reply:
What would those checks look like? I would assume that if someone is capable of injecting a `crypto` global, they are also capable of passing whatever duck typing checks we have in place. Unless it's accident rather than malice, in which case `crypto.getRandomValues` would fail anyway